embulk-input-postgresql/src/main/java/org/embulk/input/postgresql/PostgreSQLInputConnection.java

```diff
@@ -27,15 +28,23 @@ public PostgreSQLInputConnection(Connection connection, String schemaName, Optio
     }
 
     @Override
-    protected BatchSelect newBatchSelect(PreparedQuery preparedQuery,
-            List<ColumnGetter> getters,
-            int fetchRows, int queryTimeout) throws SQLException
+    protected BatchSelect newBatchSelect(PreparedQuery preparedQuery, List<ColumnGetter> getters, int fetchRows,
+                                         int queryTimeout, OptionalInt limit) throws SQLException
     {
         String query = "DECLARE cur NO SCROLL CURSOR FOR " + preparedQuery.getQuery();
+
         List<JdbcLiteral> params = preparedQuery.getParameters();
 
         logger.info("SQL: " + query);
         PreparedStatement stmt = connection.prepareStatement(query);
+
+        String fetchSql = "FETCH FORWARD "+fetchRows+" FROM cur";
+
+        if (limit.isPresent()) {
+            stmt.setMaxRows(limit.getAsInt());
+            fetchSql = "FETCH FORWARD "+limit.getAsInt()+" FROM cur";
+        }
```

Comment:
How about making it like below so that `fetchSql` is determined just once?

```suggestion
        final String fetchSql;
        if (limit.isPresent()) {
            stmt.setMaxRows(limit.getAsInt());
            fetchSql = "FETCH FORWARD "+limit.getAsInt()+" FROM cur";
        }
        else {
            fetchSql = "FETCH FORWARD "+fetchRows+" FROM cur";
        }
```

Reply:
Thanks @dmikurube I updated it.